core/templates/components/forms/custom-forms-directives/image-uploader.directive.ts

```diff
@@ -51,19 +53,53 @@ angular.module('oppia').directive('imageUploader', [
             return 'This file is not recognized as an image.';
           }
 
-          if (!file.type.match('image/jpeg') &&
-            !file.type.match('image/gif') &&
-            !file.type.match('image/jpg') &&
-            !file.type.match('image/png') &&
-            !file.type.match('image/svg\\+xml')) {
-            return 'This image format is not supported.';
+          var imageTypeMapping = {
+            jpeg: {
+              format: 'image/jpeg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            jpg: {
+              format: 'image/jpg',
+              fileType: /jp(e?)g$/,
+              fileExtension: /\.jp(e?)g$/,
+            },
+            gif: {
+              format: 'image/gif',
+              fileType: /gif$/,
+              fileExtension: /\.gif$/,
+            },
+            png: {
+              format: 'image/png',
+              fileType: /png$/,
+              fileExtension: /\.png$/,
+            },
+            svg: {
+              format: 'image/svg\\+xml',
+              fileType: /svg\+xml$/,
+              fileExtension: /\.svg$/,
+            }
+          };
+          var imageHasInvalidFormat = true;
+          for (var i = 0; i < scope.getAllowedImageFormats().length; i++) {
+            var imageType = scope.getAllowedImageFormats()[i];
+            if (!imageTypeMapping.hasOwnProperty(imageType)) {
+              return (
+                imageType + ' is not in the list of allowed image formats.');
+            }
+            if (file.type.match(imageTypeMapping[imageType].format)) {
+              imageHasInvalidFormat = false;
+              if (
```

Comment:
I still don't understand what is going on here. Why do we need to check that file.type matches both "format" and "fileType"? Can these be collapsed into a single regex/check?

Don't just follow the previous code blindly. Try to understand the logic.

Reply:
Done.

Reply:
Please explain, in detail, the logic behind your reasoning to remove the regex you removed, and why doing so does not change the definition of "validity".

Reply:
The current condition for file format is not a regex and it is direct string matching which is more concrete. Initially I thought that the addition check was done for a particular reason but now I think that it was redundant code. So I think that removing it won't reduce the validity check.